src/common/containers/repository-setup.js

```diff
@@ -4,13 +4,21 @@ import { connect } from 'react-redux';
 import { withRouter } from 'react-router';
 
 import { createWebhook } from '../actions/webhook';
-import { requestBuilds } from '../actions/request-snap-builds';
+import { requestBuilds } from '../actions/snap-builds';
 import { Message } from '../components/forms';
 import Spinner from '../components/spinner';
 
 import styles from './container.css';
 
 class RepositorySetup extends Component {
+  componentWillReceiveProps(nextProps) {
```

Comment:
Moved redirecting out of `render` to make sure `render` always returns valid component and to trigger redirect even before anything is rendered.